intellij/src/saros/intellij/filesystem/IntellijFileImplV2.java

```diff
@@ -0,0 +1,235 @@
+   * @see VirtualFile#delete(Object)
+   */
+  private void deleteInternal() throws IOException {
+    VirtualFile virtualFile = getVirtualFile();
+
+    if (virtualFile == null || !virtualFile.exists()) {
+      log.debug("Ignoring file deletion request for " + this + " as it does not exist.");
+
+      return;
+    }
+
+    if (virtualFile.isDirectory()) {
+      throw new IOException("Failed to delete " + this + " as it is not a file");
+    }
+
+    virtualFile.delete(this);
+  }
+
+  @Override
+  @Nullable
+  public String getCharset() throws IOException {
+    VirtualFile virtualFile = getVirtualFile();
+
+    if (!existsInternal(virtualFile)) {
+      throw new FileNotFoundException(
+          "Could not obtain charset for " + this + " as it does not exist or is not valid");
+    }
+
+    return virtualFile.getCharset().name();
+  }
+
+  @Override
+  public void setCharset(@Nullable String charset) throws IOException {
+    if (charset == null) {
+      return;
+    }
+
+    VirtualFile virtualFile = getVirtualFile();
+
+    if (!existsInternal(virtualFile)) {
+      throw new FileNotFoundException(
+          "Could not set charset for " + this + " as it does not exist or is not valid");
+    }
+
+    virtualFile.setCharset(Charset.forName(charset));
+  }
+
+  @Override
+  @NotNull
+  public InputStream getContents() throws IOException {
+    VirtualFile virtualFile = getVirtualFile();
+
+    if (!existsInternal(virtualFile)) {
+      throw new FileNotFoundException(
+          "Could not obtain contents for " + this + " as it does not exist or is not valid");
+    }
+
+    return virtualFile.getInputStream();
+  }
+
+  @Override
+  public void setContents(@Nullable InputStream input) throws IOException {
+    FilesystemRunner.runWriteAction(
+        (ThrowableComputable<Void, IOException>)
+            () -> {
+              setContentsInternal(input);
+
+              return null;
+            },
+        ModalityState.defaultModalityState());
+  }
+
+  /**
+   * Sets the content of the file.
+   *
+   * <p>Writes the content in chunks of size {@link #BUFFER_SIZE}.
+   *
+   * @param input an input stream to write into the file
+   * @throws IOException if the file does not exist or could not be written to
+   */
+  private void setContentsInternal(@Nullable InputStream input) throws IOException {
+    VirtualFile virtualFile = getVirtualFile();
+
+    if (!existsInternal(virtualFile)) {
+      throw new FileNotFoundException(
+          "Could not set contents of " + this + " as it does not exist or is not valid.");
+    }
+
+    OutputStream out = virtualFile.getOutputStream(this);
+
+    InputStream in = input == null ? new ByteArrayInputStream(new byte[0]) : input;
+
+    byte[] buffer = new byte[BUFFER_SIZE];
+
+    int read;
+
+    try {
+
+      while ((read = in.read(buffer)) != -1) out.write(buffer, 0, read);
+
+    } finally {
+      IOUtils.closeQuietly(out);
+      IOUtils.closeQuietly(in);
+    }
```

Comment:
please use try-with-resources, you  could also use apache commons for the copy

```suggestion
    try (InputStream in = input == null ? new ByteArrayInputStream(new byte[0]) : input;
        OutputStream out = virtualFile.getOutputStream(this); ) {
      IOUtils.copy(in, out);
    }
```

Reply:
I copied this from the previous implementation. Done.